File: ps-backend/src/main/java/org/commonground/ps/backendapi/model/Page.java
package org.commonground.ps.backendapi.model;

import java.util.List;

import org.commonground.ps.backendapi.model.template.EndpointT;

public interface Page {
    void setId(Long id);
    Long getId();

    void setName(String name);
    String getName();

    void setLayoutType(String layoutType);
    String getLayoutType();

    void setPageType(PageType pageType);
    PageType getPageType();

    void setPageComponents(List<PageComponents> pageComponents);
    List<PageComponents> getPageComponents();

    void setPageEndpoints(List<EndpointT> endpoints);
    List<EndpointT> getPageEndpoints();

    List<PageButton> getButtonsLeft();
    void setButtonsLeft(List<PageButton> pageButtons);
    List<PageButton> getButtonsRight();
    void setButtonsRight(List<PageButton> pageButtons);
}

File: ps-frontend/projects/ps-lib/src/model/page.ts
import { PageConfig } from "./domain-type-config";
import { PageButton } from "./page-button";
import { PageOverviewTemplate } from "./page-overview-template";
import { PageType } from "./page-type";

export class Page {
    public id: number;
    public name: string;
    public layoutType: string;
    public pageType: PageType;
    public buttonsLeft: PageButton[];
    public buttonsRight: PageButton[];
    public pageOverviewTemplate?: PageOverviewTemplate[];
    public pageConfig: PageConfig;
}
  